src/utils.ts

```diff
@@ -182,3 +185,81 @@ export function forceDataPrefix(url: string, mime = 'image/png') {
   if (url.startsWith('data:')) return url
   return `data:${mime};base64,` + url
 }
+
+async function getSdInfo(ctx: Context, config: Config, path: string): Promise<any> {
+  return ctx.http.axios(trimSlash(config.endpoint) + path, {
+    method: 'GET',
+    timeout: config.requestTimeout,
+    headers: {
+      ...config.headers,
+    },
+  }).then(res => res.data)
+}
+
+export async function getCkptList(ctx: Context, config: Config): Promise<StableDiffusionWebUI.ModelListResponse> {
+  return getSdInfo(ctx, config, '/sdapi/v1/sd-models')
+}
+
+export async function getLoraList(ctx: Context, config: Config): Promise<StableDiffusionWebUI.LoraListResponse> {
+  return getSdInfo(ctx, config, '/sdapi/v1/loras')
+}
+
+export async function getLycoList(ctx: Context, config: Config): Promise<StableDiffusionWebUI.LycoListResponse> {
+  return getSdInfo(ctx, config, '/sdapi/v1/lycos')
+}
+
+export async function getEmbeddingsList(ctx: Context, config: Config): Promise<StableDiffusionWebUI.EmbeddingsListResponse> {
+  return getSdInfo(ctx, config, '/sdapi/v1/embeddings')
+}
+
+export async function getHypernetworksList(ctx: Context, config: Config): Promise<StableDiffusionWebUI.HypernetworkListResponse> {
+  return getSdInfo(ctx, config, '/sdapi/v1/hypernetworks')
+}
+
+async function filterFilesByFilenameNExts(dirname: string, filename: string, exts: string[]): Promise<string> {
+  try {
+    if (!fs.existsSync(dirname)) {
+      console.error(`${dirname} doesn't exist`)
+      return null
+    }
+    const files = await fs.promises.readdir(dirname)
+    const filteredFiles = files.filter(file => {
+      const extension = file.split('.').pop().toLowerCase()
+      return file.startsWith(filename) && exts.some(ext => ext === extension)
+    })
+    return filteredFiles.length > 0 ? path.join(dirname, filteredFiles[0]) : null
+  } catch (err) {
+    console.error(err)
+    return null
+  }
+}
+
+export async function loadModelPreviewImage(modelName: string, modelPath: string): Promise<{ img: Buffer; mime: string }> {
+  const dirname = path.dirname(modelPath)
+  const img = await filterFilesByFilenameNExts(dirname, modelName, ['png', 'jpg'])
+  if (!img) {
+    console.error(`Unable to load model ${modelName}'s preview image`)
+    return null
+  }
+  return {
+    img: fs.readFileSync(img),
+    mime: path.extname(img) === '.jpg' ? 'image/jpg' : 'image/png',
```

Comment:
JPEG's `extname` can also be `.jpeg`, besides, it should be `image/jpeg` rather than `image/jpg`.
So it would be better to do `path.extname(img) === '.png' ? 'image/png' : 'image/jpeg'.

Ref: https://www.rfc-editor.org/rfc/rfc3745